### DISCUSSION TITLE
SSH input error

### DISCUSSION THREAD

**Discussion by pixply** [ghostty-org/ghostty#3671] [Feature Requests, Ideas] [answered]
When I log in to a ssh server that has oh-my-zsh, the input is completely broken. This is for example my try to exit. The last line ist just the word "exit" typed in. Backspace doesnt work, etc.

<img width="301" alt="image" src="https://github.com/user-attachments/assets/82c5fc42-0e83-447a-a2c4-ecfac40d3187" />

**Comment by RomanAverin** [comment, marked as answer]
https://ghostty.org/docs/help/terminfo#ssh

**Comment by RhetTbull** [comment]
The instructions at https://ghostty.org/docs/help/terminfo#ssh didn't work for me (macOS Ventura on Intel Mac ssh-ing to another Mac) so I'm posting this here in case others run across the same issue:

1. The path to homebrew's infocmp is only valid on Apple Silicon Macs. On Intel, the infocmp installed with ncurses will be at `/usr/local/opt/ncurses/bin/infocmp` **not** `/opt/homebrew/opt/ncurses/bin/infocmp` as listed in the docs.
2. I got error `tic: Can't open -` when trying to pipe the results of `infocmp` to `ssh`.  Rather than try to debug this as I've never used `infocmp` or `tic`, I just manually copied the terminfo file and ran tic on the remote machine using these steps:

- `brew install ncurses`
- `/usr/local/opt/ncurses/bin/infocmp -x > ghostty.terminfo`
- `scp ghostty.terminfo YOUR-USER@YOUR-SERVER:~/`
- `ssh YOUR-USER@YOUR-SERVER`
- `tic -x ~/ghostty.terminfo`

Remove the terminfo files when done.

**Reply by kevinrobayna** [reply]
The instructions are now updated:

```shell
brew install ncurses
"$(eval "brew --prefix")/opt/ncurses/bin/infocmp" -x | ssh YOUR-SERVER -- tic -x -
```

is the way to solve this (specially if you're on a mac)

**Comment by msehnout** [comment]
I fixed it with the SSH config from docs, but applied to all hosts:
```
~ $ head -3 .ssh/config
# Default configuration for all hosts
Host *
	SetEnv TERM=xterm-256color
```
